docs/how-tos/scaling.rst

```diff
@@ -0,0 +1,143 @@
+.. _scaling:
+
+Scaling your deployment
+***********************
+
+By default, the `Aspects Tutor plugin`_ deploys single nodes for these services:
+
+* :ref:`quick-start-ralph` or :ref:`quick-start-vector`
+* Clickhouse: ephemeral single node cluster
+* Superset: shares Open edX's MySQL and Redis instances
+
+Most deployments will benefit greatly from scaling horizontally and/or vertically, especially when running Aspects. Aspects is fed by Open edX event data, which is triggered by user actions on the site, and Open edX can generate a lot of event data. The initial processing of this event data occurs in platform plugins, so scaling your LMS workers speeds this processing time. And the faster events can be processed, the quicker you will see them appear in Aspects' dashboards and charts.
+
+And all production deployments will need a `persistent Clickhouse cluster`_.
+
+Preparing the LMS workers
+=========================
+
+Before deploying Aspects, we recommend :ref:`event_bus`.
+
+Or, if you'd rather use celery, we recommend boosting the number of LMS workers to prepare the celery queue for the high-volume of Aspects tasks.
+
+These plugins can be used to configure and enable autoscaling for the LMS and CMS. See their READMEs for details:
+
+* `tutor-contrib-pod-autoscaling`_ : for single instance deployments
+* `tutor-contrib-grove`_ : for multiple instance deployments
+
+We also recommend configuring the LMS to use the high-priority celery queue for Aspects tasks (`platform-plugin-aspects event sinks`_, `event-routing-backends xAPI tasks`_). This leaves the low-priority queue clear for other LMS tasks.
+
+.. code-block:: yaml
+
+  TBD -- implement configurable queue for Aspects tasks
+
+Scaling Ralph
+=============
+
+Without scaling, Ralph can be a bottleneck in the Aspects data pipeline. If you do not need an LRS at all, you can disable Ralph and use :ref:`quick-start-vector` instead.
+
+But if you will ever need an LRS, it's better to start with Ralph enabled and autoscaled.
+
+Ralph runs CPU-intensive operations, so we recommend scaling mainly on CPU:
+
+.. code-block:: yaml
+
+  TBD -- implement RALPH autoscaling
+
+.. _scaling-clickhouse:
+
+Scaling Clickhouse
+==================
+
+We recommend using a Clickhouse service provider to manage your production cluster.
```

Comment:
@Ian2012 @bmtcril I've been playing with deploying aspects using our [Grove](https://gitlab.com/opencraft/dev/grove) deployment system, and it was trivial to get it running with the default single-node Clickhouse deployment and persistent volume: https://superset.jill-aspects.staging.do.opencraft.hosting

I haven't tried using Harmony yet.

But if we're recommending using a hosted CH in production, do we need to add support for a scalable CH deployment to Harmony for Aspects v1? I'm happy to try, but I'm worried about maintaining something if no one is using it.

Reply:
Yeah, a lot of folks prefer to run their own for cost or data privacy reasons so we should support scaling for it. I think we should be pretty up front about complicated scaling configurations being at your own risk since there's no way we can support them all.

Reply:
Cool, that can be documented as part of https://github.com/openedx/openedx-aspects/issues/80.